apps/src/templates/teacherDashboard/SectionSelectionDropdown.jsx

```diff
@@ -0,0 +1,133 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import color from '@cdo/apps/util/color';
+import i18n from '@cdo/locale';
+import PopUpMenu from '../../lib/ui/PopUpMenu';
+import SectionSelectionMenuItem from './SectionSelectionMenuItem';
+import {sectionForDropdownShape} from './shapes';
+import SmallChevronLink from '@cdo/apps/templates/SmallChevronLink';
+
+const styles = {
+  addNewSection: {
+    borderTop: `1px solid ${color.charcoal}`,
+    paddingTop: 16,
+    paddingBottom: 8,
+    paddingLeft: 12,
+    paddingRight: 12
+  }
+};
+
+export default class SectionSelectionDropdown extends Component {
+  static propTypes = {
+    sections: PropTypes.arrayOf(sectionForDropdownShape).isRequired,
+    selectedSection: PropTypes.object,
+    selectSection: PropTypes.func
+  };
+
+  state = {
+    selectedSection: this.props.selectedSection,
```

Comment:
nit: please name this `initialSelectedSection` or similar. otherwise, bugs where you accidentally reference `props.selectedSection` can be tricky to track down. More context here: https://medium.com/@justintulk/react-anti-patterns-props-in-initial-state-28687846cc2e (although the react docs they link to appear to have since changed)